jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -486,6 +492,11 @@ List<Image> getCachedBaseImages()
       }
 
       ManifestTemplate manifest = Verify.verifyNotNull(manifestAndConfigFound.get().getManifest());
+      // Verify all layers described in manifest are present in cache
```

Comment:
(optional) there was quite a bit of code duplication in this method even in the past between the section that deals with general and platform-based manifest processing. Could there be an opportunity for refactoring common logic out into a helper method?

Reply:
Ah this is a great suggestion - will refactor this and see if I can make the methods cleaner to read.